src/runners/rfq_maker_balance_cache_runner.ts

```diff
@@ -0,0 +1,169 @@
+// Cache the query results to reduce reads from the DB
+let MAKER_TOKEN_CACHE: ResultCache<MakerBalanceChainCacheEntity[]>;
+const _getMakerTokensAsync = async (connection: Connection) => {
+    if (!MAKER_TOKEN_CACHE) {
+        logUtils.log('updating cache');
+        MAKER_TOKEN_CACHE = createResultCache<any[]>(
+            () =>
+                connection
+                    .getRepository(MakerBalanceChainCacheEntity)
+                    .createQueryBuilder('maker_balance_chain_cache')
+                    .select(['maker_balance_chain_cache.tokenAddress', 'maker_balance_chain_cache.makerAddress'])
+                    .getMany(),
+            TEN_MINUTES_MS,
+        );
+    }
+    return (await MAKER_TOKEN_CACHE.getResultAsync()).result;
+};
+
+function _splitValues(makerTokens: MakerBalanceChainCacheEntity[]): BalancesCallInput {
+    const functionInputs: BalancesCallInput = { addresses: [], tokens: [] };
+
+    return makerTokens.reduce(({ addresses, tokens }, makerToken) => {
+        return {
+            addresses: addresses.concat(makerToken.makerAddress as string),
+            tokens: tokens.concat(makerToken.tokenAddress as string),
+        };
+    }, functionInputs);
+}
+
+async function _getBalanceCheckerContractInterfaceAsync(
+    provider: ethers.providers.JsonRpcProvider,
+): Promise<ethers.Contract> {
+    return new ethers.Contract(BALANCE_CHECKER_ADDRESS, BALANCE_CHECKER_ABI, provider);
+}
+
+async function _getErc20BalancesAsync(
+    balanceCheckerContractInterface: ethers.Contract,
+    balancesCallInput: BalancesCallInput,
+): Promise<string[]> {
+    // due to gas contraints limit the call to 1K balance checks
+    const addressesChunkedArray = _.chunk(balancesCallInput.addresses, MAX_BALANCE_CHECKS_PER_CALL);
+    const tokensChunkedArray = _.chunk(balancesCallInput.tokens, MAX_BALANCE_CHECKS_PER_CALL);
+
+    const balances = await Promise.all(
+        addressesChunkedArray.map(async (addresses, i) => {
+            return balanceCheckerContractInterface.functions.balances(addresses, tokensChunkedArray[i]);
+        }),
+    );
+
+    const balancesFlattened = Array.prototype.concat.apply([], balances);
+
+    return balancesFlattened.map((bal: any) => bal.toString());
+}
+
+async function _updateErc20BalancesAsync(
+    balancesCallInput: BalancesCallInput,
+    balances: string[],
+    connection: Connection,
+    updateTime: Date,
+): Promise<void> {
+    const toSave = await Promise.all(
+        balancesCallInput.addresses.map(async (addr, i) => {
+            const dbEntity = new MakerBalanceChainCacheEntity();
+
+            dbEntity.makerAddress = addr;
+            dbEntity.tokenAddress = balancesCallInput.tokens[i];
+            dbEntity.balance = new BigNumber(balances[i]);
+            dbEntity.timeOfSample = updateTime;
+
+            return dbEntity;
+        }),
+    );
+
+    await connection.getRepository(MakerBalanceChainCacheEntity).save(toSave);
```

Comment:
What does the query look like for this statement? Have you tried running `TypeORM` in debug mode and testing this out>

Reply:
`advanced-console`